custom/icds_reports/management/commands/dashboard_usage_report.py

```diff
@@ -0,0 +1,244 @@
+
+
+def cache_to_file(cache_name):
+    def _outer(fn):
+        @wraps(fn)
+        def _inner(*args, **kwargs):
+            data = _get_from_file(cache_name)
+            if not data:
+                data = fn(*args, **kwargs)
+                _write_to_file(cache_name, data)
+            return data
+        return _inner
+    return _outer
+
+
+def _get_from_file(filename):
+    if os.path.exists(filename):
+        print(f'Fetching data from file: {filename}')
+        with open(filename, 'r') as f:
+            reader = csv.reader(f)
+            return [
+                r[0] if len(r) == 1 else r for r in list(reader)
+            ]
+
+
+def _write_to_file(filename, rows):
+    print(f'Writing {len(rows)} to file {filename}')
+    with open(filename, 'w') as f:
+        writer = csv.writer(f)
+        writer.writerows([
+            r if isinstance(r, list) else [r] for r in rows
+        ])
+
+
+class Command(BaseCommand):
+    required_fields = ['state_id', 'state_name', 'district_id', 'district_name', 'block_id', 'block_name']
+    location_types = ['state_id', 'district_id', 'block_id']
+    user_level_list = ['State', 'District', 'Block']
+    location_test_fields = ['state_is_test', 'district_is_test', 'block_is_test', 'supervisor_is_test',
+                            'awc_is_test']
+
+    roles = {
+        '.nod': 'Nodal Officer',
+        '.ncd': 'Consultant(Nutrition & Child Development)',
+        '.bcc': 'Consultant (Behaviour Change Communication & Capacity Building)',
+        '.sdc': 'Consultant (Social Development & Community Mobilization)',
+        '.mne': 'Consultant (Monitoring & Evaluation and Decentralized Planning)',
+        '.fm': 'Consultant (Financial Management)',
+        '.shd': 'Consultant (Procurement) (State Helpdesk)',
+        '.pa': 'Project Associate',
+        '.acc': 'Accountant',
+        '.cta': 'Central Training Agent',
+        '.dhd': 'District Coordinator (District Helpdesk)',
+        '.dpa': 'District Project Assistant',
+        '.dc': 'District Collector',
+        '.bhd': 'Block Coordinator (Block Helpdesk)',
+        '.bpa': 'Block Project Assistant',
+        '.dpo': 'District Programme Officer (DPO)',
+        '.cdpo': 'Child Development Project Officer (CDPO)',
+
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'start_date',
+            type=date_type,
+            help='The start date (inclusive). format YYYY-MM-DD'
+        )
+        parser.add_argument(
+            'end_date',
+            type=date_type,
+            help='The end date (exclusive). format YYYY-MM-DD'
+        )
+
+    def handle(self, start_date, end_date, **options):
+        domain = 'icds-cas'
+
+        print(f'fetching users data')
+        all_usernames = get_all_usernames_by_domain(domain)
+
+        dashboard_uname_rx = re.compile(r'^\d*\.[a-zA-Z]*@.*')
+
+        usernames = [username for username in all_usernames if dashboard_uname_rx.match(username)]
```

Comment:
Also if you directly pull the usage report in starting itself you will get this username list in that only. you wont need to filter here again.

Reply:
changed it